### PR TITLE
Allow building on Visual Studio 2022

### LINKED ISSUE #13033: Support building with Visual Studio 2022
### Is your feature request related to a problem? Please describe.
Visual Studio 2022 is to be released next week. NVDA builds with Visual Studio 2019.

### Describe the solution you'd like
Support building with Visual Studio 2022. We can decide to support 2019 and up, but as soon as appveyor supports 2022, it makes sense to drop 2019 support altogether.

### Additional context
Blocked by a new SCons release. Visual Studio 2022 support is in master but is awaiting a new release.

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
Closes #13033

### Summary of the issue:
NVDA now builds with Visual Studio 2019. Version 2022 was released recently.

### Description of how this pull request fixes the issue:
1. Updates SCons to version 4.3.0.
2. Allows building with Visual Studio 2019 and newer.
3. Updates the readme with download url's that are likely to be pretty stable in the near and more distant future.

### Testing strategy:
Tested locally and on appveyor.

### Known issues with pull request:
It looks like MS made it pretty difficult to get to the downloads of 2019 through their website without logging in, therefore I'm linking to the executables directly using their short urls.

### Change log entries:
For Developers
- Compiling NVDA dependencies with Visual Studio 2022 (17.0) is now supported. For development and release builds, Visual Studio 2019 is still used. (#13033)
- Updated SCons to version 4.3.0.

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

**Comment by lukaszgo1** [comment]
While I certainly agree that allowing to build with VS 2022 is pretty important as some people may want to use the newest version available I'm not convinced preventing compilation with 2019 is a good idea as per the several reasons:

1. VS 2022 does not offer us any visible benefits at this stage, and since it has been released only 4 days ago it is much less tested.
2. [System requirements for Visual Studio 2022](https://docs.microsoft.com/en-us/visualstudio/releases/2022/system-requirements) are pretty unreasonable essentially preventing anyone running 32-bit versions of Windows or LTSC versions of Windows 10 (which are still in support for a long time) from contributing to NVDA without a good reason.
3. Historically I believe we tried to allow building NVDA under all versions of Windows which NVDA supports - I've  no idea if this has been ever written down as a rule or was a coincidence.
4. If some corporation would like to contribute to NVDA Visual Studio is the only one of our dependencies which cannot be used for free potentially necessitating a purchase even though previous version is still perfectly suitable for building.

There would undoubtedly come a time when still allowing to build with VS 2019  would hold us back in some way and at this point we certainly should drop this support, but for now there is no real need in my opinion.
I personally believe this PR should be changed so that VS 2019 and up are accepted for compilation, with a mention in the readme that VS 2022 is recommended but compilation can be performed with 2019 and up.

**Comment by leonardder** [comment]
> 4. If some corporation would like to contribute to NVDA Visual Studio is the only one of our dependencies which cannot be used for free potentially necessitating a purchase even though previous version is still perfectly suitable for building.

I don't understand this one. You can perfectly install several editions of Visual Studio side by side, including enterprise and community.
> 
> I personally believe this PR should be changed so that VS 2019 and up are accepted for compilation, with a mention in the readme that VS 2022 is recommended but compilation can be performed with 2019 and up.

I'm happy to do that, though I think it makes more sense to stick to Visual Studio 2019 on appveyor in this case and yet allow building with 2022. This ensures that, when we need to make a change that breaks Visual Studio 2019, we'll be notified automatically.

**Comment by lukaszgo1** [comment]
> I don't understand this one. You can perfectly install several editions of Visual Studio side by side, including enterprise and community.

Sure you can. The problem is that if you are a enterprise there are several restrictions for using Community version of VS as mentioned in its EULA, therefore it is likely  you cannot use Community edition legally in that case.

> I'm happy to do that, though I think it makes more sense to stick to Visual Studio 2019 on appveyor in this case and yet allow building with 2022. This ensures that, when we need to make a change that breaks Visual Studio 2019, we'll be notified automatically.

Makes sense.

**Comment by leonardder** [comment]
I updated the pr description.

File: nvdaHelper/archBuild_sconscript [hunk 1]
```diff
@@ -73,9 +73,9 @@ def COMProxyDllBuilder(env,target,source,proxyClsid):
 	return proxyDll
 env.AddMethod(COMProxyDllBuilder,'COMProxyDll')
 
-# We only support compiling with MSVC 14.2 (2019)
-if not env.get('MSVC_VERSION')=='14.2':
-	raise RuntimeError("Visual C++ 14.2 (Visual Studio 2019) not found")
+# We only support compiling with MSVC 14.2 (2019) or newer
+if not env.get('MSVC_VERSION') >= '14.2':
```
Comment (michaelDCurran) [inline nvdaHelper/archBuild_sconscript:77]: I'm not sure this check will do what you think if there happened to be a 14.10, though I do realize this is very unlikely.

Comment (leonardder) [reply nvdaHelper/archBuild_sconscript:77]: I guess we could fix this.

```suggestion
if not env.get('MSVC_VERSION') or env.get('MSVC_VERSION').split(".") < (14, 2):
```

**Comment by leonardder** [comment]
scons 4.3.0 is there, so we're ready to rock.

**Review by michaelDCurran — APPROVED** [review]

**Comment by leonardder** [comment]
This was in fact reverted by #13387

**Pull request merged**